File: docs/code-examples/annotation_context_segmentation.cpp
// Log a segmentation image with annotations.

#include <rerun.hpp>

#include <algorithm>

int main() {
    auto rec = rerun::RecordingStream("rerun_example_annotation_context_connections");
    rec.connect("127.0.0.1:9876").throw_on_failure();

    // create an annotation context to describe the classes
    rec.log_timeless(
        "segmentation",
        rerun::AnnotationContext({
            rerun::AnnotationInfo(1, "red", rerun::Rgba32(255, 0, 0)),
            rerun::AnnotationInfo(2, "green", rerun::Rgba32(0, 255, 0)),
        })
    );

    // create a segmentation image
    const int HEIGHT = 8;
    const int WIDTH = 12;
    std::vector<uint8_t> data(WIDTH * HEIGHT, 0);
    for (auto y = 0; y < 4; ++y) {                   // top half
        std::fill_n(data.begin() + y * WIDTH, 6, 1); // left half
    }
    for (auto y = 4; y < 8; ++y) {                       // bottom half
        std::fill_n(data.begin() + y * WIDTH + 6, 6, 2); // right half
    }

    rec.log("segmentation/image", rerun::SegmentationImage(WIDTH, HEIGHT, std::move(data)));
}

File: docs/snippets/all/annotation_context_segmentation.py
"""Log a segmentation image with annotations."""

import numpy as np
import rerun as rr

rr.init("rerun_example_annotation_context_segmentation", spawn=True)

# Create a simple segmentation image
image = np.zeros((200, 300), dtype=np.uint8)
image[50:100, 50:120] = 1
image[100:180, 130:280] = 2

# Log an annotation context to assign a label and color to each class
rr.log("segmentation", rr.AnnotationContext([(1, "red", (255, 0, 0)), (2, "green", (0, 255, 0))]), static=True)

rr.log("segmentation/image", rr.SegmentationImage(image))
